File: src/main/java/task/java/leetcode75/lvl1/IsSubsequence.java
package task.java.leetcode75.lvl1;

public class IsSubsequence {
    public boolean isSubsequence(String s, String t) {
        if (s == null || t == null) return false;
        if (s.isBlank()) return true;
        if (s.isBlank() && t.isBlank()) return true;

        int counter = 0;
        for (int i = 0; i < t.length() && counter < s.length(); i++) {
            if (s.charAt(counter) == t.charAt(i)) {
                counter++;
            }
        }
        return counter == s.length();
    }
}

File: src/main/kotlin/task/kotlin/leetcode75/lvl1/IsSubsequence.kt
package task.kotlin.leetcode75.lvl1

// https://leetcode.com/problems/is-subsequence
// time: O(n)
// space: O(1)

class IsSubsequence {
    fun isSubsequence(s: String, t: String): Boolean {
        if (s.isBlank()) return true
        if (t.isBlank() && s.isBlank()) return true
        if (t.isBlank() && s.isNotBlank()) return false

        var sCounter = 0

        for (i in t.indices) {
            if (sCounter == s.length) return true
            if (s[sCounter] == t[i])
                sCounter++
        }
        return sCounter == s.length
    }
}